web/source/dom/domEventHandlers.ts

```diff
@@ -734,6 +675,156 @@ namespace com.keyman.dom {
+        const maxY = touchY;
+        const minY = touchY - yRow;
+
+        // Performs a binary search for a valid caret based on the y-position.
+        // cp:  the previously-set caret position.
+        for(let iLoop=0; iLoop<16; iLoop++) {
+          // Break the binary search if our final search window is extremely small.
+          if(cpMax - cpMin <= 1) {
+            break;
+          }
+          const y=dom.Utils.getAbsoluteY(caret)-dy;  //top of caret
+
+          if(y > maxY && cp > cpMin /*&& cp != cpMax*/) {
+            // If caret's prior placement is below (after) the touch's y-pos...
+            cpMax=cp;  // new max position
+            cp=Math.round((cp+cpMin)/2); // guess the halfway mark
+          } else if(y < minY && cp < cpMax /*&& cp != cpMin*/) {
+            // If caret's prior placement is above (before) the touch's y-pos - 1 row height...
+            cpMin=cp; // new min posiiton
+            cp=Math.round((cp+cpMax)/2); // guess the halfway mark
+          } else { // the y-position lines up; cp is within the target row.
+            break;
+          }
+          // Actively set our caret to the determined matching y-position.
+          target.setTextCaret(cp);  // mutates `caret`'s position
+        }
+
+        // cpMin, cpMax are either absolute bounds (beginning/end of context)
+        // or are outside of the target row at this point.
+
+        // Because of situations with new-lines, we still need to ensure the caret's actually within the target row.
+        while(dom.Utils.getAbsoluteY(caret)-dy > maxY && cp > cpMin) {
+          target.setTextCaret(--cp); // mutates `caret`'s position
+          cpMin = cp; // Old location was not on the same line.  It's free; may as well take it.
+          // If this block executed, cpMin will be 'tight' when complete.
+        }
+
+        while(dom.Utils.getAbsoluteY(caret)-dy < minY && cp < cpMax) {
+          target.setTextCaret(++cp); // mutates `caret`'s position
+          cpMax = cp; // Old location was not on the same line.  It's free; may as well take it.
+          // If this block executed, cpMax will be 'tight' when complete.
+        }
+
+        // The caret itself should now lie within the target row.  The start of the row must be at or before
+        // the caret, while the end of the row must be at or after it.
+        //
+        // To properly reuse the InputHTMLElement-based x-coord search, we need the bounds to be TIGHT after
+        // this comment's containing block finishes. As in, we need [cpMin, cpMax] === [start of row, end of row].
+        //
+        // Note:  in certain scenarios, one of the bounds can be VERY far away at this point.
+        // Say, if a new touch event wants to move the caret nearby the previous location, as the
+        // bound on the opposite side may not have gotten overwritten even once so far in this call!
+        // Therefore... more binary search.
```

Comment:
I'm not sure I understand this comment?

Reply:
Thanks. Can you move this commentary into the code -- it'll help future maintainers.

Reply:
Is this - taken from the followup, #6566 - sufficient?

https://github.com/keymanapp/keyman/blob/7990cdc1dd06f9c5adb8a35db4332542a2c9c3c6/web/source/dom/touchAliasElement.ts#L490-L498

That PR was how I chose to handle this aspect mentioned above:

> I'll see if I can tidy things up to make the code itself more concise and more intuitive to follow; it just turned out that there was a lot of initially-unexpected nuance that had to be handled... 

------

Other relevant altered/new comments added in that PR:

https://github.com/keymanapp/keyman/blob/7990cdc1dd06f9c5adb8a35db4332542a2c9c3c6/web/source/dom/touchAliasElement.ts#L454

https://github.com/keymanapp/keyman/blob/7990cdc1dd06f9c5adb8a35db4332542a2c9c3c6/web/source/dom/touchAliasElement.ts#L470-L475

These comments are designed to highlight that the first binary search is not focused on finding cpMin and cpMax.   It's about making guarantees on one value that will greatly help us to find cpMin and cpMax.  We just take what we can get on the way, but note that we don't guarantee we can get anything but that one value - a single index that happens to lie on the correct vertical line - and what it represents.

Reply:
Yep, these comments LGTM